### PR TITLE
Reintroduce ebpf newtools

### PR DESCRIPTION
**Pull request by nplanel**

### CODE REVIEW

File: flow/ebpf_test.go [hunk 1]
```diff
@@ -42,6 +42,6 @@ func TestEBPFFlow(t *testing.T) {
 	table.Stop()
 
 	if len(expired) != 3 {
-		t.Errorf("Table should have 3 flows, got : %d", table.table)
+		t.Errorf("Table should have 3 flows, got : %d ", table.table.Len())
```
Comment (safchain) [inline flow/ebpf_test.go]: this is racy

Comment (safchain) [reply flow/ebpf_test.go]: that is a nit

**Comment by nplanel** [comment]
run skydive-cdd-overview-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by nplanel** [comment]
run skydive-scale-tests

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

File: flow/ebpf_test.go [hunk 2]
```diff
@@ -42,6 +42,9 @@ func TestEBPFFlow(t *testing.T) {
 	table.Stop()
 
 	if len(expired) != 3 {
-		t.Errorf("Table should have 3 flows, got : %d", table.table)
+		t.Errorf("Table should have expired 3 flows, got : %d", len(expired))
+	}
+	if table.table.Len() > 0 {
```
Comment (safchain) [inline flow/ebpf_test.go]: it is racy, you can read table.Len here. I think you can simply remove this

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

**Comment by nplanel** [comment]
run skydive-functional-hw-tests

File: flow/ebpf.go [hunk 3]
```diff
@@ -283,31 +285,76 @@ func (ft *Table) newFlowFromEBPF(ebpfFlow *EBPFFlow, key uint64) ([]uint64, []*F
...
+						portA := int64(ebpfFlow.KernFlow.transport_layer.port_src)
+						portB := int64(ebpfFlow.KernFlow.transport_layer.port_dst)
+						cmp = portA > portB
+					}
+				}
+			}
+		}
+	}
+	return cmp
+}
```
Comment (safchain) [inline flow/ebpf.go]: add an empty line after this one

File: flow/probes/ebpf.go [hunk 4]
```diff
@@ -66,14 +80,24 @@ type EBPFProbesHandler struct {
 	wg  sync.WaitGroup
 }
 
+func (p *EBPFProbe) swapPage() {
+	key := uint32(C.FLOW_PAGE)
+	writeInPage := int64(p.flowPage)
+
+	p.cmap.Put(key, writeInPage)
+
+	if p.flowPage == 1 {
```
Comment (safchain) [inline flow/probes/ebpf.go]: p.flowPage = 1 - p.flowPage

**Comment by nplanel** [comment]
run tests

**Comment by nplanel** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by nplanel** [comment]
run skydive-cdd-overview-tests

**Comment by nplanel** [comment]
run skydive-compile-tests

**Comment by nplanel** [comment]
run skydive-cdd-overview-tests

**Comment by nplanel** [comment]
run skydive-cdd-overview-tests

**Comment by nplanel** [comment]
run skydive-cdd-overview-tests

**Comment by nplanel** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-cdd-overview-tests

**Pull request merged**